ciao-controller/internal/datastore/datastore.go

```diff
@@ -2280,3 +2257,61 @@ func (ds *Datastore) UnMapExternalIP(address string) error {
 
 	return nil
 }
+
+func (ds *Datastore) generateCNCIWorkload() {
+	// generate the CNCI workload.
+	config := `---
+#cloud-config
+users:
+  - name: cloud-admin
+    gecos: CIAO Cloud Admin
+    lock-passwd: false
+    passwd: $6$rounds=4096$w9I3hR4g/hu$AnYjaC2DfznbPSG3vxsgtgAS4mJwWBkcR74Y/KHNB5OsfAlA4gpU5j6CHWMOkkt9j.9d7OYJXJ4icXHzKXTAO.
+    sudo: ALL=(ALL) NOPASSWD:ALL
+...
+`
+	cpus := payloads.RequestedResource{
+		Type:      payloads.VCPUs,
+		Value:     4,
+		Mandatory: false,
+	}
+
+	mem := payloads.RequestedResource{
+		Type:      payloads.MemMB,
+		Value:     128,
```

Comment:
Adding an api one way or another to configure the cnci has been my plan all along btw, I just thought I'd do it in another PR. But I can do it in this one, no problem. it would also resolve the issue of not being able to pass ssh keys to the cnci for debug (and those map failure tests in verify.sh)